### PR TITLE
Add support for --profile for selecting IAM credentials

### PR DESCRIPTION
**Pull request by stshan**
Related Issue: https://github.com/awslabs/aws-sam-local/issues/27

Adds --profile as a new flag for start-api and invoke to allow user to specify specific profiles if needed, instead of always using the default profile

### CODE REVIEW

**Comment by resios** [comment]
Why not set the `AWS_PROFILE` environment variable as instructed by: http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default

**Comment by sanathkr** [comment]
`--profile` is also the standard AWS CLI way of specifying profile. Tools other than AWS CLI that source creds have generally exposed this flag to customers in addition to AWS_PROFILE. It is a good to have

File: main.go [hunk 1]
```diff
@@ -108,6 +108,10 @@ func main() {
 							Usage:  "Optional. Specify whether SAM should skip pulling down the latest Docker image. Default is false.",
 							EnvVar: "SAM_SKIP_PULL_IMAGE",
 						},
+						cli.StringFlag{
```
Comment (sanathkr) [inline main.go]: Notes for future: common options like profile must be abstracted into an `args` object and passed down to Invoke/StartApi for processing.

**Comment by cam8001** [comment]
Is this flag in the current version of sam-cli? It doesn't seem to show up in 0.15 from Brew:

```
$ sam --version                                                                                                                                                                
SAM CLI, version 0.15.0

$ sam --profile datalake                                                                                                                                                        
Error: no such option: --profile
```

**Comment by Zanndorin** [comment]
@cam8001 It is but at the end of the command
`sam local start-api --profile devAdmin`

https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-local-invoke.html

**Pull request merged**